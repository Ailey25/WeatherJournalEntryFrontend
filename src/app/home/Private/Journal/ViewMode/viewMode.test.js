import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import 'babel-polyfill';

import { ViewModeContainer } from './index';

configure({ adapter: new Adapter() })

describe('ViewModeContainer', () => {
  it('renders correctly', () => {
    const match = {
      params: {
        id: 'placeholderId'
      }
    };
    const props = {
      journalList: [],
      match: match,
    }
    const wrapper = shallow(
      <ViewModeContainer {...props} />
    );

    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  });
});

// assume getJournal() works
describe('ViewModeContainer', () => {
  let props = {};
  let wrapper = {};

  beforeEach(() => {
    props = {
      journalList: [],
      history: [],
      match: {
        params: {
          id: 'placeholderId'
        }
      },
    };
    wrapper = shallow(
      <ViewModeContainer {...props} />
    );
  });

  // states are set based on journal from getJournal()
  // mock the states
  it('renders title that exists correctly', () => {
    let mockState = { title: 'fake title' };
    wrapper.setState({ ...mockState });

    expect(wrapper.find('#journalHeader').text()).toBe(mockState.title);
  });

  it('renders title that DOESN\'T exists correctly', () => {
    let mockState = { title: '' };
    wrapper.setState({ ...mockState });
    const expectedTitle = 'No title!';

    expect(wrapper.find('#journalHeader').text()).toBe(expectedTitle);
  });

  it('renders entry that exists correctly', () => {
    let mockState = { entry: 'fake entry here' };
    wrapper.setState({ ...mockState });

    const entryComponentProps = wrapper.find('#entry').props();
    expect(entryComponentProps.value).toBe(mockState.entry);
  });

  it('renders entry that DOESN\'T exists correctly', () => {
    let mockState = { entry: '' };
    wrapper.setState({ ...mockState });
    const expectedEntry = 'No entry!';

    const entryComponentProps = wrapper.find('#entry').props();
    expect(entryComponentProps.value).toBe(expectedEntry);
  });
});
