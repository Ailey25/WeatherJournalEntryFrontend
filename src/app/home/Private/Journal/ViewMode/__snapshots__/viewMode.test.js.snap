// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewModeContainer renders correctly 1`] = `
<styled.div>
  <Styled(Link)
    to="/journals"
  >
    Back to journal entries
  </Styled(Link)>
  <h2
    id="journalHeader"
  >
    No title!
  </h2>
  <hr />
  <div
    className="rowCenter"
  >
    <withRouter(Connect(WeatherStampContainer))
      id="placeholderId"
    />
  </div>
  <div
    className="rowCenter"
  >
    <Styled(Styled(styled.textarea))
      cols="100"
      disabled={true}
      id="entry"
      rows="30"
      value="No entry!"
    />
  </div>
</styled.div>
`;
